cas-server-support-osf/src/main/java/io/cos/cas/authentication/handler/support/OpenScienceFrameworkPrincipalFromRequestRemoteUserNonInteractiveCredentialsAction.java

```diff
@@ -457,12 +491,12 @@ protected PrincipalAuthenticationResult notifyRemotePrincipalAuthenticated(
                 throw new RemoteUserFailedLoginException("Invalid Status Code from OSF API Endpoint");
             }
 
-            // return the username for the credential build.
+            // Return the username for the credential build.
             return new PrincipalAuthenticationResult(username, institutionId);
-        } catch (final JOSEException | IOException | ParserConfigurationException | TransformerException e) {
```

Comment:
All exceptions were handled in one place here before, which are now split up and moved to handle independent functional parts.

* `ParserConfigurationException` and `TransformerException`: catch attributes parsing and normalization errors
* `JOSEException`: catch JWE/JWT encryption errors
* `IOException`: catch HTTP request/response errors